src/trackers/process-crawl.js

```diff
@@ -44,16 +44,17 @@ async function processCrawl() {
     try {
         bar = new Progress('Process crawl [:bar] :percent', {width: 40, total: await reader.length()})
 
-        let sites = []
+        const sitesQueue = []
         for await (const siteData of reader.iterator()) {
-            if (sites.length >= sharedData.config.parallelism) {
-                // wait for batch to finish before reading more site data
-                await Promise.allSettled(sites)
```

Comment:
This was processing batches of 100 and waiting till all sites in a batch are processed. Instead of waiting for the slowest entry to finish we can move on as soon as any of the sites finishes processing.

Reply:
I think this was intentional to avoid being blocked or rate limited from doing too many CNAME look ups. I'll try it out though.